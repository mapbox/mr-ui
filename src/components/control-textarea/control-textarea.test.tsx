import React from 'react';
import ControlTextarea from './control-textarea';
import { screen, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('ControlTextarea', () => {
  describe('basic', () => {
    const mockOnChange = jest.fn();
    const props = {
      id: 'testinput',
      label: 'Basic',
      onChange: mockOnChange
    } as const;

    test('renders as expected', () => {
      const { baseElement } = render(<ControlTextarea {...props} />)
      expect(baseElement).toMatchSnapshot();
    });

    test('onChange is called with input id and value', async () => {
      render(<ControlTextarea {...props} />)
      await userEvent.type(screen.getByTestId('testinput-textarea'), 'f');

      await waitFor(() => {
        expect(mockOnChange).toHaveBeenCalledTimes(1);
      });

      await waitFor(() => {
        expect(mockOnChange).toHaveBeenCalledWith('f', 'testinput');
      });
    });
  });

  describe('disabled', () => {
    const mockOnChange = jest.fn();
    const props = {
      id: 'testinput',
      label: 'basic',
      onChange: mockOnChange,
      disabled: true
    } as const;

    test('renders as expected', () => {
      const { baseElement } = render(<ControlTextarea {...props} />)
      expect(baseElement).toMatchSnapshot();
    });

    test('onChange is called with input id and value', async () => {
      render(<ControlTextarea {...props} />)
      await userEvent.type(screen.getByTestId('testinput-textarea'), 'steve');

      await waitFor(() => {
        expect(mockOnChange).toHaveBeenCalledTimes(0);
      });
    });
  });

  describe('all options', () => {
    const props = {
      id: 'testinput',
      label: 'All options',
      validationError: 'oh no!',
      noAuto: true,
      optional: true,
      autoFocus: true,
      aside: <span>Aside text</span>,
      placeholder: 'Some cool placeholder text',
      value: 'Some cool default value',
      themeControlTextarea: 'w-full border--pink round-bold px12 py12',
      themeControlWrapper: 'bg-red px12 py12',
      themeLabel: 'txt-s',
      onChange: jest.fn()
    } as const;

    test('renders as expected', () => {
      const { baseElement } = render(<ControlTextarea {...props} />)
      expect(baseElement).toMatchSnapshot();
    });
  });
});
