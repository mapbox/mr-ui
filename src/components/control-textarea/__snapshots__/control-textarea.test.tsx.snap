// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlTextarea all options renders as expected 1`] = `
<body>
  <div>
    <div
      class="relative bg-red px12 py12"
    >
      <div>
        <label
          class="inline-block txt-s"
          for="testinput"
        >
          All options
           
          <span
            class="txt-normal"
          >
            (optional)
          </span>
        </label>
        <span
          class="inline-block ml12"
        >
          <span>
            Aside text
          </span>
        </span>
      </div>
      <textarea
        aria-invalid="true"
        aria-required="false"
        autocapitalize="none"
        autocorrect="off"
        class="w-full border--pink round-bold px12 py12"
        data-testid="testinput-textarea"
        id="testinput"
        name="testinput"
        placeholder="Some cool placeholder text"
        spellcheck="false"
      >
        Some cool default value
      </textarea>
      <div
        role="alert"
      >
        <div
          class="bg txt-s mt3 py6 px12 bg-red-faint round border border--red color-red"
          data-testid="testinput-error"
          id="testinput-error"
        >
          oh no!
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ControlTextarea basic renders as expected 1`] = `
<body>
  <div>
    <div
      class="relative "
    >
      <div>
        <label
          class="inline-block txt-s txt-bold mb6"
          for="testinput"
        >
          Basic
           
        </label>
      </div>
      <textarea
        aria-required="true"
        class="textarea hmin120"
        data-testid="testinput-textarea"
        id="testinput"
        name="testinput"
      />
      <div
        role="alert"
      />
    </div>
  </div>
</body>
`;

exports[`ControlTextarea disabled renders as expected 1`] = `
<body>
  <div>
    <div
      class="relative "
    >
      <div>
        <label
          class="inline-block txt-s txt-bold mb6"
          for="testinput"
        >
          basic
           
        </label>
      </div>
      <textarea
        aria-required="true"
        class="textarea hmin120"
        data-testid="testinput-textarea"
        disabled=""
        id="testinput"
        name="testinput"
      />
      <div
        role="alert"
      />
    </div>
  </div>
</body>
`;
